scripts/sign_encrypt.py

```diff
@@ -488,10 +606,82 @@ def display(self):
                 if len(img) != img_size:
                     raise Exception("Unexpected img size: got {}, expected {}"
                                     .format(len(img), img_size))
-        else:
-            raise Exception("Unsupported image type: {}".format(img_type))
+            offs += img_size
 
 
+        offs = 0
+        while offs < len(self.inf):
+            if offs > 0:
+                # name_size is the previous subkey header
+                name_img = self.inf[offs:offs + name_size]
+                print('  next name:  "{}"'.format(name_img_to_str(name_img)))
+                offs += name_size
+                print('Next header at offset: {} (0x{:x})'
+                      .format(offs, offs))
+
+            shdr = self.inf[offs:offs + SHDR_SIZE]
+            [magic, img_type, img_size, algo_value, digest_len,
+             sig_len] = struct.unpack('<IIIIHH', shdr)
```

Comment:
s/digest_len/hash_size/ and s/sig_len/sig_size/ to match the C definition and the text below

Reply:
OK, I'll do that in a new commit on top of this since it touches quite a lot.